presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -416,8 +416,11 @@ public void addPreparedStatement(String key, String value)
         requireNonNull(key, "key is null");
         requireNonNull(value, "value is null");
 
+        if (session.getPreparedStatements().containsKey(key)) {
+            throw new PrestoException(ALREADY_EXISTS, "Prepared statement already exists: " + key);
+        }
         String previousValue = addedPreparedStatements.putIfAbsent(key, value);
-        if (previousValue != null || session.getPreparedStatements().containsKey(key)) {
+        if (previousValue != null) {
```

Comment:
Could you ever get into a situation where the if statement above doesn't get triggered (i.e. the key is not already in the hash table) but you get null here?